plugins/woocommerce-admin/client/products/fields/variations/variations.tsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { Card, Spinner } from '@wordpress/components';
+import {
+	EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME,
+	ProductVariation,
+} from '@woocommerce/data';
+import { ListItem, Sortable, Tag } from '@woocommerce/components';
+import { useContext } from '@wordpress/element';
+import { useParams } from 'react-router-dom';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { CurrencyContext } from '../../../lib/currency-context';
+import { getProductStockStatus } from '../../utils/get-product-stock-status';
+import './variations.scss';
+
+export const Variations: React.FC = () => {
+	const { productId } = useParams();
+	const context = useContext( CurrencyContext );
+	const { formatAmount, getCurrencyConfig } = context;
+	const { isLoading, variations } = useSelect( ( select ) => {
+		const { getProductVariations, hasFinishedResolution } = select(
+			EXPERIMENTAL_PRODUCT_VARIATIONS_STORE_NAME
+		);
+		return {
+			isLoading: ! hasFinishedResolution( 'getProductVariations', [
+				{
+					product_id: productId,
+				},
+			] ),
+			variations: getProductVariations< ProductVariation[] >( {
+				product_id: productId,
+			} ),
+		};
+	} );
+
+	if ( ! variations || isLoading ) {
```

Comment:
Note that the current behavior is odd because some products don't have variations.  [Automatic generation of variations](https://github.com/woocommerce/woocommerce/issues/35778) will address this and hopefully guarantee that a variation always exists if any options exist.